cpp/arcticdb/util/error_code.hpp

```diff
@@ -30,19 +30,22 @@ enum class ErrorCategory : BaseType {
     SCHEMA = 4,
     STORAGE = 5,
     SORTING = 6,
+    USER_INPUT = 7
     // NEW CATEGORIES MUST ALSO BE ADDED TO python_module.cpp:register_error_code_ecosystem
 };
 
 // FUTURE(GCC9): use magic_enum
-inline std::unordered_map<ErrorCategory, const char*> get_error_category_names() {
-    return {
+inline decltype(auto) get_error_category_names() {
+    static const std::unordered_map<ErrorCategory, const char*> error_category_map{
         {ErrorCategory::INTERNAL, "INTERNAL"},
         {ErrorCategory::NORMALIZATION, "NORMALIZATION"},
         {ErrorCategory::MISSING_DATA, "MISSING_DATA"},
         {ErrorCategory::SCHEMA, "SCHEMA"},
         {ErrorCategory::STORAGE, "STORAGE"},
-        {ErrorCategory::SORTING, "SORTING"}
+        {ErrorCategory::SORTING, "SORTING"},
+        {ErrorCategory::USER_INPUT, "USER_INPUT"}
```

Comment:
Please leave a `,` at the end where possible. Same for the `ERROR_CODES` macro below